pkg/util/intern/intern.go

```diff
@@ -0,0 +1,70 @@
+package intern
+
+import (
+	"sync"
+	"unsafe"
+
+	pq "github.com/parquet-go/parquet-go"
+)
+
+type Interner struct {
+	mtx sync.RWMutex
+	m   map[string][]byte
+}
+
+func New() *Interner {
+	return &Interner{m: make(map[string][]byte)}
+}
+
+func (i *Interner) UnsafeClone(v *pq.Value) pq.Value {
+	a := *(*aValue)(unsafe.Pointer(v))
+	switch v.Kind() {
+	case pq.ByteArray, pq.FixedLenByteArray:
+		a.ptr = addressOfBytes(i.internBytes(v.ByteArray()))
+		return *(*pq.Value)(unsafe.Pointer(&a))
+	default:
+		return *v
```

Comment:
does this incur an extra value copy?

Reply:
Yes, but we pass `pq.Value` as values (🥁) in `Entries`.

Reply:
But if we take a pointer why don't we return a pointer? We would not have to clone in that case.

Reply:
Aren't we cloning values unnecessarily? For any non-byte array/fixed len byte array by cloning here?

Reply:
Yes, but values are used everywhere. Changing this to pointers is a big change in the codebase, that I'd prefer to keep out from this PR.

FWIW, [`v.Clone()`](https://github.com/grafana/tempo/blob/main/pkg/parquetquery/iters.go#L936) is/was creating a copy of `pq.Value` too. This is unchanged in this PR.

Reply:
should the line be `v.Clone()` to make it consistent with the way it used to be? i'm honestly not sure.

Reply:
For all other types (int, float, etc) the struct doesn't contain external pointers so a copy of the contents is fine.  That's what Clone() does:  https://github.com/parquet-go/parquet-go/blob/main/value.go#L792    Avoiding it is fine because it will incur another type check unnecessarily.

Reply:
Fine with this leaving as is in the PR, but it does feel safer to call `.Clone()` in case that internal behavior changes.